filip/clients/ngsi_v2/cb.py

```diff
@@ -1165,49 +1179,50 @@ def post_subscription(self,
 
-        sub_hash = subscription.model_dump_json(include={'subject', 'notification'})
+        sub_hash = subscription.model_dump_json(include={"subject", "notification"})
         for ex_sub in existing_subscriptions:
-            if sub_hash == ex_sub.model_dump_json(include={'subject', 'notification'}):
+            if sub_hash == ex_sub.model_dump_json(include={"subject", "notification"}):
                 self.logger.info("Subscription already exists")
                 if update:
                     self.logger.info("Updated subscription")
                     subscription.id = ex_sub.id
                     self.update_subscription(subscription)
                 else:
-                    warnings.warn(f"Subscription existed already with the id"
-                                  f" {ex_sub.id}")
+                    warnings.warn(
+                        f"Subscription existed already with the id" f" {ex_sub.id}"
+                    )
                 return ex_sub.id
 
         params = {}
         if skip_initial_notification:
-            version = self.get_version()['orion']['version']
-            if parse_version(version) <= parse_version('3.1'):
-                params.update({'options': "skipInitialNotification"})
+            version = self.get_version()["orion"]["version"]
+            if parse_version(version) <= parse_version("3.1"):
+                params.update({"options": "skipInitialNotification"})
             else:
                 pass
-            warnings.warn(f"Skip initial notifications is a deprecated "
-                          f"feature of older versions <=3.1 of the context "
-                          f"broker. The Context Broker that you requesting has "
-                          f"version: {version}. For newer versions we "
-                          f"automatically skip this option. Consider "
-                          f"refactoring and updating your services",
-                          DeprecationWarning)
-
-        url = urljoin(self.base_url, 'v2/subscriptions')
+            warnings.warn(
+                f"Skip initial notifications is a deprecated "
+                f"feature of older versions <=3.1 of the context "
+                f"broker. The Context Broker that you requesting has "
+                f"version: {version}. For newer versions we "
+                f"automatically skip this option. Consider "
+                f"refactoring and updating your services",
+                DeprecationWarning,
+            )
+
+        url = urljoin(self.base_url, "v2/subscriptions")
         headers = self.headers.copy()
-        headers.update({'Content-Type': 'application/json'})
+        headers.update({"Content-Type": "application/json"})
         try:
             res = self.post(
                 url=url,
                 headers=headers,
-                data=subscription.model_dump_json(exclude={'id'},
-                                                  exclude_unset=True,
```

Comment:
Why the `exclude_unset` is also removed here?

Reply:
@djs0109 its because if it was excluded. The default values if not explitly set would be removed as well. Hence, the user would expect the default value will be sent to the plattform.